model/schema/allele.yaml

```diff
@@ -235,25 +233,41 @@ classes:
       note_dto:
         required: true
 
-  AlleleSynonymSlotAnnotation:
-    is_a: SlotAnnotation
+  AlleleSymbolSlotAnnotation:
+    description: >-
+      The one current symbol for the allele.
+    is_a: NameSlotAnnotation
     slots:
       - single_allele
-      - synonym
-      - synonym_scope
     slot_usage:
       single_allele:
         required: true
-      synonym:
+      name_type:
+        any_of:
+          - equals_string: symbol
+
+
+  AlleleFullNameSlotAnnotation:
+    description: >-
+      The one current full name for the allele.
+    is_a: NameSlotAnnotation
+    slots:
+      - single_allele
+    slot_usage:
+      single_allele:
         required: true
+      name_type:
+        any_of:
+          - equals_string: full_name
```

Comment:
I'm wondering: I know the name_type_name is a DTO class slot and so the range is "string" and therefore we can use the "equals_string" assertion, but I wonder if we can apply the same constraint for "name_type" where the range is actually 'VocabularyTerm'?

Reply:
Changed to note